File: abc392a/abc392a.ml
open Printf
open Scanf

let solve a =
  match List.sort compare a with
  | x :: y :: z :: _ when x * y = z -> "Yes"
  | _ -> "No"

let () =
  List.init 3 (fun _ -> scanf "%d " @@ fun x -> x) |> solve |> printf "%s\n"

File: abc392a/abc392a.hs
{-# LANGUAGE OverloadedStrings #-}

import Data.List(sort)
import Data.Bool (bool)

main :: IO ()
main = solve <$> f >>= putStrLn
  where
    f = map read <$> words <$> getLine

solve :: [Int] -> String
solve = bool "No" "Yes" . f . sort
  where
    f [a, b, c] = a * b == c
